[7.2.0] Unrecoverable error while evaluating node 'SINGLE_EXTENSION_EVAL:ModuleExtensionId

### Description of the bug:


Hi,

we've observed the following error on Windows after upgrading to 7.2.0 and moving `rules_rust` over to bzlmod (both happened at the same time).
The error is spurious.

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'SINGLE_EXTENSION_EVAL:ModuleExtensionId{bzlFileLabel=@@rules_rust~//crate_universe:extension.bzl, extensionName=crate, isolationKey=Optional.empty}' (requested by nodes 'SINGLE_EXTENSION:ModuleExtensionId{bzlFileLabel=@@rules_rust~//crate_universe:extension.bzl, extensionName=crate, isolationKey=Optional.empty}')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:550)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:414)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: Can't get the number of an unknown enum value.
	at com.google.devtools.build.lib.server.FailureDetails$ExternalDeps$Code.getNumber(FailureDetails.java:57724)
	at com.google.devtools.build.lib.server.FailureDetails$ExternalDeps$Builder.setCode(FailureDetails.java:58182)
	at com.google.devtools.build.lib.bazel.bzlmod.ExternalDepsException.<init>(ExternalDepsException.java:37)
	at com.google.devtools.build.lib.bazel.bzlmod.ExternalDepsException.withCauseAndMessage(ExternalDepsException.java:51)
	at com.google.devtools.build.lib.bazel.bzlmod.SingleExtensionEvalFunction$RegularRunnableExtension.run(SingleExtensionEvalFunction.java:948)
	at com.google.devtools.build.lib.bazel.bzlmod.SingleExtensionEvalFunction.compute(SingleExtensionEvalFunction.java:194)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:461)
	... 7 more
BAZEL FAILED
```

### Which category does this issue belong to?

_No response_

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


N/A

Some part of our repository is open-source though, including the offending module extensions definition, see: https://github.com/github/codeql (current main)

### Which operating system are you running Bazel on?


Windows Server 2019

### What is the output of `bazel info release`?


7.2.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?


_No response_

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.


The error is spurious, so it's hard to bisect.

### Have you found anything relevant by searching the web?

no

### Any other information, logs, or outputs that you want to share?

_No response_

--- a/src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java
+++ b/src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java
@@ -946,7 +946,7 @@ public class SingleExtensionEvalFunction implements SkyFunction {
           } catch (IOException e1) {
             ExternalDepsException externalDepsException =
                 ExternalDepsException.withCauseAndMessage(
-                    ExternalDeps.Code.UNRECOGNIZED,
+                    ExternalDeps.Code.EXTERNAL_DEPS_UNKNOWN,
                     e1,
                     "Failed to clean up module context directory");
             throw new SingleExtensionEvalFunctionException(

